### ISSUE TITLE
Concurrent Streams

### ISSUE THREAD

**Issue by dceldran** [martinbjeldbak/acestream-http-proxy#34]
Hi, 

It's any chance to add the feature to have more than one stream at same time with same docker container?

When i try to start more than one stream at the same time, i have random disconnections in all the streams.

Thanks!!

**Comment by martinbjeldbak** [comment]
hi @dceldran this should be achievable by running multiple docker containers as well, any reason why you can't do this?

**Comment by dceldran** [comment]
Hi @martinbjeldbak,

Thanks for the fast answer.

My usecase is the following:

I've created an .m3u file with some acestream channels. I like to use this .m3u in home devices like some tvs, smartphones... and it's possible that there are 2 or 3 concurrent streams at the same time.

I've been thinking in another possibilities like having some containers, for example 3, and generate 3 .m3u files. The problem is that the maintaining of this file is more complex because i have to maintain 3 m3u files instead of one.

The other possibility is to use some containers and add a load balancer (like traefik for example) to balance the traffic on all containers. But honestly i don't know if this is going to work or not.

Thanks!!

**Comment by bijfy** [comment]
@dceldran how did you make m3u out of acestream.  Kindly share some details?

**Comment by dceldran** [comment]
Hi @bijfy,

For example if you have the acestream-http-proxy container in 192.168.10.20:6878, you can create an .m3u file with content:

```
#EXTM3U
#EXTINF:-1,1. Acestream Channel 1 group-title="ACESTREAM" tvg-id="Acestream Channel 1"
http://192.168.10.20:6878/ace/getstream?id=XXXX
#EXTINF:-1,2. Acestream Channel 2 group-title="ACESTREAM" tvg-id="Acestream Channel 2"
http://192.168.10.20:6878/ace/getstream?id=YYYY
```

After that, if you can't load the m3u file directly to your client app, you can create a docker container with httpd server and put the file inside, so you can download from: `http://192.168.10.20/playlist.m3u` for example.

Ofc this only works in local network, if you want to use this for example with your smartphone outside your local network you need to open ports or use some proxy like NPM or using an vpn.

And the problem with this is that you can only use once at a time. If you want to see any channel in your TV and another channel in your smartphone at the same time, you need to create two containers of acestream-http-proxy with different ports and create two .m3u files and load that different m3u files in the different clients you have. And if the acestream ids changes, you need to update more than one .m3u file.

I really like this project because it's easy to see acestream in all devices you need. You only need some client that accepts m3u and voila. I really hope that will be viable to implement the concurrent streams. Thanks for this project :)

**Comment by bijfy** [comment]
Thank you for the info.

**Comment by bijfy** [comment]
services:
  ace22:
     image: ghcr.io/martinbjeldbak/acestream-http-proxy:latest
     container_name: ace22
     network_mode: "container:glu3"
     environment:
      - ALLOW_REMOTE_ACCESS=yes
      - HTTP_PORT=6822
     restart: always


 I have 4 containers with different ports through my vpn container.

**Comment by iercan** [comment]
@dceldran

If you provide different pid for each channel, you can watch more than one channel at a time. 

For example 
```
#EXTM3U
#EXTINF:-1,1. Acestream Channel 1 group-title="ACESTREAM" tvg-id="Acestream Channel 1"
http://192.168.10.20:6878/ace/getstream?id=XXXX&pid=1
#EXTINF:-1,2. Acestream Channel 2 group-title="ACESTREAM" tvg-id="Acestream Channel 2"
http://192.168.10.20:6878/ace/getstream?id=YYYY&pid=2
```

**Comment by dceldran** [comment]
Hi @iercan 

Finally i'm using another project, it's a proxy between the client that is playing an stream and the acestream engine and is capable to automatically create new pids for new streams and reuse existing streams if you need two clients playing same channel for example.

Thanks

**Comment by bijfy** [comment]
@dceldran which project are you using ?

**Comment by martinbjeldbak** [comment]
@bijfy this project I suspect: https://github.com/Javinator9889/acexy

**Comment by dceldran** [comment]
Hi @bijfy 

Like @martinbjeldbak says, the project is https://github.com/Javinator9889/acexy.

**Comment by iercan** [comment]
Nice work by acexy developer. @martinbjeldbak maybe we could add this feature to the this project too.

**Comment by martinbjeldbak** [comment]
agree @iercan maybe we can work with them to accelerate any features on this project's backlog. I like the idea of it being built on top of our project and seamlessly extends functionality at a higher level

**Comment by diegofercri** [comment]
Hi to everyone and specially to @dceldran ,

Adding a feature like this and some good documentation could be great, cause I can't get Acexy working as I really want with Tivimate

**Comment by dceldran** [comment]
Hi @diegofercri,

The Acexy project is in early stage development, there are bugs to fix. My usecase is to use it with Threadfin (xTeve fork) and my problem is that if the acestream channel is offline, the connection remains open and the proxy became unresponsive, for example if i have 3 channels to see the same content and i try to see one that is offline, when i try to see other channel also don't works because the proxy is unresponsive, so i have to restart docker container and try again. With the next release the author implements a timeout to avoid this but i don't know when a new release comes out. Maybe you have a similar problem?

**Comment by diegofercri** [comment]
That's great @dceldran I'm also using Threadfin and Tivimate.

I don't know if it's exactly the same problem, I have managed to get all the channels to appear in Tivimate but I have not been able to play any content.

It would be great if you could help me a bit.

Should we move this thread to the Acexy repository or somewhere else?

**Comment by dceldran** [comment]
Hi @diegofercri Personally i'm never used Tivimate so i don't know if i can help you or not.

If you have errors on acexy proxy with Tivimate i suppose that the right place to post the issue is in acexy repository, but if tivimate tries to access to all channels before play one of them, maybe you have the same problem than me. You can try to do a playlist with only 2 or 3 channels that you know that are working and try with Tivimate.

**Comment by Javinator9889** [comment]
> Hi [@diegofercri](https://github.com/diegofercri),
> 
> The Acexy project is in early stage development, there are bugs to fix. My usecase is to use it with Threadfin (xTeve fork) and my problem is that if the acestream channel is offline, the connection remains open and the proxy became unresponsive, for example if i have 3 channels to see the same content and i try to see one that is offline, when i try to see other channel also don't works because the proxy is unresponsive, so i have to restart docker container and try again. With the next release the author implements a timeout to avoid this but i don't know when a new release comes out. Maybe you have a similar problem?

Hello there 👋🏼 

Just to add some news, the Acexy project is continuously evolving and this problem should have been fixed :)

As always, feel free to fill in any bug or issue report. Right now I'm addressing bugs at a pace that allows me to release Acexy within two weeks

https://github.com/Javinator9889/acexy

**Comment by dceldran** [comment]
> > Hi [@diegofercri](https://github.com/diegofercri),
> > The Acexy project is in early stage development, there are bugs to fix. My usecase is to use it with Threadfin (xTeve fork) and my problem is that if the acestream channel is offline, the connection remains open and the proxy became unresponsive, for example if i have 3 channels to see the same content and i try to see one that is offline, when i try to see other channel also don't works because the proxy is unresponsive, so i have to restart docker container and try again. With the next release the author implements a timeout to avoid this but i don't know when a new release comes out. Maybe you have a similar problem?
> 
> Hello there 👋🏼
> 
> Just to add some news, the Acexy project is continuously evolving and this problem should have been fixed :)
> 
> As always, feel free to fill in any bug or issue report. Right now I'm addressing bugs at a pace that allows me to release Acexy within two weeks
> 
> https://github.com/Javinator9889/acexy

Yeah i’m using acexy for a few weeks and it works great! Thanks!!

**Comment by naijaping** [comment]
Hi, thanks for this lovely project, I find it very easy for me to spin in docker than acexy.  can someone please confirm if concurrent streams features is coming to acestream-http-proxy anytime soon.

Many thanks

**Comment by martinbjeldbak** [comment]
Hi @naijaping thanks for your kind message. You should be able to run acexy in docker alongside the acestream-http-proxy image. Are you having troubles with this?

**Comment by naijaping** [comment]
@martinbjeldbak , thanks for your swift response, I will give acexy a go second time. When I tried it, the buffering  was too much and loads of streams dropout unlike acestream-http-proxy standalone which just works well for me apart from the concurrent Streams limitations.

Many thanks for your hard work on these wonderful project.

**Comment by Javinator9889** [comment]
> @martinbjeldbak , thanks for your swift response, I will give acexy a go second time. When I tried it, the buffering  was too much and loads of streams dropout unlike acestream-http-proxy standalone which just works well for me apart from the concurrent Streams limitations.
> 
> Many thanks for your hard work on these wonderful project.

Hey! Just wanted to say everything should be fully customizable on Acexy. If you find any troubles or problems when using it, please create an issue so we can all benefit :)

BR
